suse_migration_services/prechecks/kernels.py

```diff
@@ -54,24 +56,50 @@ def multiversion_and_multiple_kernels():
                         'Please ensure it is set as:\n'
                         "'multiversion.kernels = latest,running'")
         elif set(kernel_multi_version_settings.split(',')) != {'running', 'latest'}:
-            log.warning('The config option multiversion.kernels is not set '
-                        'correctly in /etc/zypp/zypp.conf. It is currrently '
-                        'set as \n%s.\nPlease ensure it is set as:\n'
-                        "'multiversion.kernels = latest,running'",
-                        kernel_multi_version_settings)
+            log.warning("The config option multiversion.kernels is not set "
+                        "correctly in /etc/zypp/zypp.conf. It is currrently "
+                        "set as:\n'multiversion.kernels = %s'.", kernel_multi_version_settings)
+            if fix:
+                log.warning("The '--fix' option was provided, "
+                            "setting 'multiversion.kernels = latest,running' ")
+                existing = 'multiversion.kernels = ' + kernel_multi_version_settings
+                correction = 'multiversion.kernels = latest,running'
+                sed_arg = 's/' + existing + '/' + correction + '/'
+                try:
+                    Command.run(["sed", "-i", sed_arg, Defaults.get_zypp_config_path()])
+                except DistMigrationCommandException:
+                    log.error("ERROR: Unable to update /etc/zypp/zypp.conf")
+
+            else:
+                log.warning("Please ensure it is set as:\n"
+                            "'multiversion.kernels = latest,running'")
 
     kernel_type = 'kernel-default'
-    running_kernel = Command.run(["uname", "-r"])
+    running_kernel = Command.run(["uname", "-r"]).output
```

Comment:
Note in the context of the life system this will return the name of the kernel in the ISO image

Reply:
I have updated this, please review